### ISSUE TITLE
Interface to deployment tools is not consistent

### ISSUE THREAD

**Issue by rmm-fcul** [larics/assisi-python#29]
Three tools, `deploy.py`, `assisirun.py`, `collect_data.py` use the `<project>.assisi` file as input.  The `sim.py` helper for spawning CASUs in a playground instance accepts the `<project>.arena` file as input.  
Is there a strong reason for maintaining the different interface to `sim.py`? Or could it also accept the `.assisi` file, thus giving a consistent interface? (this is basically a pointer to 3 config files, including the arena file).

**Comment by dmiklic** [comment]
This seems reasonable, will do.

**Comment by dmiklic** [comment]
On second thought, I really don't see a reason for `sim.py` to accept an `.assisi` file. It requires only the info contained within the `.arena` file. To explain further, `sim.py` is not strictly a "deployment" tool. It is only a convenience tool for spawning a bunch of stuff in the simulator. It can be used with or without the rest of the deployment tools, i.e. I can see use cases where the user will make an `.arena` file without an corresponding `.assisi` file.

In short, unless there is compelling reason against, I'd vote for closing this issue with a wontfix label :)

**Comment by rmm-fcul** [comment]
My rationale is not a use-case for `sim.py` needing additional information, but rather, that the user needs to duplicate information: That is, to run a simulation that has CASUs in it, I will use `sim.py` to spawn them, `deploy.py` and then `assisirun.py` to execute the controllers.  So there is not really a way to use the simulator without having the `.assisi` project file, unless not using the bulk of the deployment toolsuite -- in which case, why would someone bother to construct a conforming `.arena` file? (not a critique of the representation, just to say that some specifics are arbitrary)

To me, it is more open to error to have to kick off some tools with file `X` (which contains a link to file `Y`) and other tools with file `Y`.  It is cleaner to have a project file that everything uses; if `Y` should become `Y'` every tool consistently uses it. 

If this takes a wontfix, then I propose to implement another wrapper around `sim.py` (projectsim.py?) that extracts the `.arena` file from the .assisi file, and then runs `sim.py`. :)

**Comment by rmm-fcul** [comment]
By the way, I had thought of a bigger change to handle this, which would be to write an `.assisi` project file parser object, that is then instanced by the entire toolsuite; ensuring a consistent interface.  However, the code is currently somewhat heterogeneous and it seems like creating work for the sake of it...  If the deployment tools are likely to expand, something along these lines might be worth considering? (regardless of the present issue)

**Comment by dmiklic** [comment]
Closed by  e25153a
